api/src/main/java/org/openmrs/module/xdssender/api/xds/MessageUtil.java

```diff
@@ -45,8 +47,9 @@
 	private XdsSenderConfig config;
 	
 	public ProvideAndRegisterDocumentSetRequestType createProvideAndRegisterDocument(byte[] documentContent,
-	        final DocumentInfo info) throws JAXBException, IOException {
-		String patientId = getPatientIdentifier(info).getIdentifier();
+	        final DocumentInfo info, Encounter encounter) throws JAXBException, IOException {
+		String patientEcidId = getEcidPatientIdentifier(info).getIdentifier();
+		String patientIsantePlusId = getIsantePusPatientIdentifier(info).getIdentifier();
```

Comment:
I think we should just the primary patient identifier here, not the iSantePlus identifier. They are the same right now for our implementation, but other implementations would just use the patient identifier that they set as the "emr.primaryIdentifierType" global property. Can you change the code to look for that global property and use it?

Reply:
Yes, I can do this.